actions/helm_stage.go

```diff
@@ -0,0 +1,105 @@
+package actions
+
+import (
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+type ghFile struct {
+	ReadCloser io.ReadCloser
+	FileName   string
+}
+
+var (
+	ourFS = getRealFileSys()
+	ourFP = getRealFilePath()
+)
+
+const (
+	// RepoFlag represents the '-repo' flag
+	RepoFlag = "repo"
+	// RefFlag represents the '-ref' flag (for specifying a SHA, branch or tag)
+	RefFlag = "ref"
+	// GHOrgFlag represents the '-ghOrg' flag
+	GHOrgFlag = "ghOrg"
+)
+
+func helmStage(ghClient *github.Client, c *cli.Context, fileNames []string, stagingSubDir string) {
+	var opt github.RepositoryContentGetOptions
+	opt.Ref = c.GlobalString(RefFlag)
+	org := c.GlobalString(GHOrgFlag)
+	repo := c.GlobalString(RepoFlag)
+
+	ghFiles, err := downloadFiles(ghClient, org, repo, &opt, fileNames)
+	if err != nil {
+		log.Fatalf("Error downloading contents of %v (%s)", fileNames, err)
+	}
+
+	stageFiles(ourFS, ghFiles, stagingPath)
+
+	if err := updateFilesWithRelease(ourFP, ourFS, deisRelease, stagingSubDir); err != nil {
+		log.Fatalf("Error updating files with release '%s' (%s)", deisRelease.Short, err)
+	}
+}
+
+func createDir(fs fileSys, dirName string) error {
+	_, err := fs.Stat(dirName)
+	if os.IsNotExist(err) {
+		if err := fs.MkdirAll(dirName, os.ModePerm); err != nil {
+			return err
+		}
+	} else {
+		return err
+	}
+	return nil
+}
+
+func downloadFiles(ghClient *github.Client, org, repo string, opt *github.RepositoryContentGetOptions, fileNames []string) ([]ghFile, error) {
+	ret := make([]ghFile, 0, len(fileNames))
+	for _, fileName := range fileNames {
+		readCloser, err := downloadContents(ghClient, org, repo, fileName, opt)
+		if err != nil {
+			return nil, err
+		}
+		ret = append(ret, ghFile{ReadCloser: readCloser, FileName: fileName})
+	}
+	return ret, nil
+}
+
+func stageFiles(fs fileSys, ghFiles []ghFile, stagingDir string) {
+	for _, ghFile := range ghFiles {
+		readCloser := ghFile.ReadCloser
+		defer readCloser.Close()
```

Comment:
this should be done at the end of each loop iteration. otherwise, all the read closers are gonna be closed at the end of the whole function